main.tf

```diff
@@ -110,7 +110,7 @@ resource "aws_db_instance" "rds" {
   engine                       = var.db_engine
   engine_version               = var.db_engine_version
   instance_class               = var.db_instance_class
-  name                         = local.db_name
+  name                         = can(regex("sqlserver", var.db_engine)) ? null : local.db_name
```

Comment:
`name` needs to be `null` for sqlserver, otherwise terraform/aws gives an error.